packages/libs/client/src/signing/keypair/createSignWithKeypair.ts

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * interface for a keypair
+ *
+ * @public
+ */
+export interface IKeypair {
+  publicKey: string;
+  secretKey: string;
+}
+
+/**
+ * interface for the `createSignWithKeypair` function {@link createSignWithKeypair}
+ *
+ * @public
+ */
+export interface ICreateSignWithKeypair {
+  /**
+   * @param key - provide the key to sign with
+   * @returns a function to sign with
+   *
+   * @example
+   * ```ts
+   * const signWithKeystore = createSignWithKeypair([keyPair, keyPair2]);
+   * const [signedTx1, signedTx2] = await signWithKeystore([tx1, tx2]);
+   * const signedTx3 = await signWithKeystore(tx3);
+   * ```
+   *
+   * @public
+   */
+  (key: IKeypair): ISignFunction;
+  /**
+   * @param keys - provide the keys to sign with
+   * @returns a function to sign with
+   *
+   *
+   * @example
+   * ```ts
+   * const signWithKeystore = createSignWithKeypair([keyPair, keyPair2]);
+   * const [signedTx1, signedTx2] = await signWithKeystore([tx1, tx2]);
+   * const signedTx3 = await signWithKeystore(tx3);
+   * ```
+   *
+   * @public
+   */
+  (keys: IKeypair[]): ISignFunction;
+}
+
+/**
+ * function to create a `signWithKeypair` function
+ * This allows you to sign subsequent transactions with the same keypair(s)
+ *
+ * @param keyOrKeys - provide the key or multiple keys to sign with
+ * @returns a function to sign with
+ *
+ * @example
+ * ```ts
+ * const signWithKeystore = createSignWithKeypair([keyPair, keyPair2]);
+ * const [signedTx1, signedTx2] = await signWithKeystore([tx1, tx2]);
+ * const signedTx3 = await signWithKeystore(tx3);
+ * ```
+ *
+ * @public
+ */
+export const createSignWithKeypair: ICreateSignWithKeypair = (keyOrKeys) => {
+  const keypairs: IKeypair[] = Array.isArray(keyOrKeys)
+    ? keyOrKeys
+    : [keyOrKeys];
+  return async function signWithKeypair(transactionList) {
+    if (transactionList === undefined) {
+      throw new Error('No transaction(s) to sign');
+    }
+
+    const isList = Array.isArray(transactionList);
+    const transactions = isList ? transactionList : [transactionList];
+
+    const signedTransactions = transactions.map((tx) => {
+      debug(`signing transaction(s): ${JSON.stringify(tx)}`);
+      const parsedTransaction = parseTransactionCommand(tx);
+      const relevantKeypairs = getRelevantKeypairs(parsedTransaction, keypairs);
+
+      if (relevantKeypairs.length === 0) {
+        throw new Error(
+          'The keypair(s) provided are not relevant to the transaction',
+        );
+      }
+
+      return signWithKeypairs(tx, relevantKeypairs);
+    });
+
+    return isList ? signedTransactions : signedTransactions[0];
+  } as ISignFunction;
+};
+
+function getRelevantKeypairs(
+  tx: IPactCommand,
+  keypairs: IKeypair[],
+): IKeypair[] {
+  const relevantKeypairs = keypairs.filter((keypair) =>
+    tx.signers.some(({ pubKey }) => pubKey === keypair.publicKey),
+  );
+  debug('relevant keypairs', relevantKeypairs);
+  return relevantKeypairs;
+}
+
+function signWithKeypairs(
```

Comment:
Shouldn't we export this function as well?
Consider I have developed my way to manage the keypairs (for example, in the dev-wallet we already did) and just need a simple function that signs the transaction with a list of keypairs.
It gives users more flexibility.

Reply:
The `createSignWithKeypair` doesn't really change that, right? Even if you had a way to manage that you can still use `const sign = createSignWithKeypair(allMyKeys)(txs)`. Also this one considers only keypairs that are already checked whether they're needed for this tx. 

Secondly, exposing this one as extra API introduces more surface that we need to maintain